client/packages/system/src/Patient/CreatePatientModal/PatientResultsTab.tsx

```diff
@@ -62,6 +64,8 @@ export const PatientResultsTab: FC<PatientPanel> = ({ patient, value }) => {
     {
       key: 'dateOfBirth',
       label: 'label.date-of-birth',
+      formatter: dateString =>
+        dateString ? localisedDate((dateString as string) || '') : '',
```

Comment:
Was just trying to figure out why `dateOfBirth` is not considered a string, and realised that columnDefinition doesn't actually enforce typing : /, I guess it's ok, i tried seeing what it would take to make it type safe, and realised it's quite a bit of work. 

Fyi type definitions would have to look something like this, [ts playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAwgFhAxgaygXigbwLACgpQD2AdhAFxQDOwATgJbEDmA3HgcAO6EXECuAtgCMINVrgC+YvKEhQAYoRr8AhsGAiAPABUAfOigAKAG7KANr3JQtASnR7q9JlNwzoAIWWU6iGIXP9iABEIADMGOmA6Em0AGigANT0MHHwoZAgQCi0xAhDFFTURAH4KYzMLCnjbNHtaBhY8SWlwaABBABMTYkQIdt9-INDwyOjdfRSCdMyqOqc2KGVEHspKRVLV3hoerOraxwaJZwB6I6gASWIiGnaRKGBCKEQaCFVoBxJGO5b2hZoaZRARBCREEACskMAsFMKAByADSGSBIPBiGAMLiJnMlhh8XK0EIwOACDSiIJyIhMPEeBOUDyNAWUEYEFI9EQ5NRcVIvTuD14lDehH40DqKBEIGasn6AmIlFalAA8mCIdoklgoABtAAKUAYJJAZK0AF0KB4vD4-NLgmFiBEosQNJq4lotYa9E0XC1YBaAlbhnaVeh5gAfKAdLo9PrewbW22jHTBr0DWUKpWolXqqYGw05KAS6BSgKUAMYAvRv1x9XZvB4RAkag6kIAOUIwAAyrNGABRP6KfRlLEUBz1XZQTEWMS1mWQuhNluNgTCGjdmi9jD9ipQPhCEQj9v7dcQazOUwQSG1pMUUtF+BIZCq9XzTB62EkCDo2n5VTqGgUGfNtsdsuvbiDEj7PlAMKcIQ750gU36-rOwDztuS49vSIFgdCEGvjBn6FD+Db-shi5AehoGpE+WGQVwuFKF+IgIf+e71KRUAYRR8yTBksJEs8b7kQQBCLMsqwERsWzQDUnGCX+LbMUwpEGOJPQAHSvrYADUhFzguIiKcpEAqVB1gCWxpkTIJVFQei0nCRAKxrFQhCbD0djSQQslIbpqErjQSnORJRlcJp2kAfs+kBap6mmexBAWVx0wwmERj8bZSz2aJg6RZJ8apIJoXyV2aH+S5hnGVAWmecRenFQZamkCZ8zsdmQA):

Reply:
yeah. I've played around a bit too - was wondering if we can extract the type from `T[key]` in the interface but no. We did spend a while trying to implement proper type checking with the column definitions and ended up with the current compromise.

Adding in type checking causes issues with the loose column definitions that we currently have: `key: keyof T | GenericColumnKey | string` - Even if we had just `keyof T` it would be better. The selection column has no data and therefore no type and the string value allows for calculations. There's a bit of work in unpicking those..

So I think to merge this issue - and create a new one for the type checking if you think it worthwhile. Could take a bit of time